[FSDP] Avoid printing incorrect warning for _get_param_to_fqns

fegin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #94494
* #94637

There exist a hack for `_get_param_to_fqns` and `_apply_to_modules`. The condition for the warning of the hack is incorrect and result in overwhelming message for users. This PR fixes the issue.

The original hack is not removed. It will once the support of DMP + FSDP is deprecated.

Differential Revision: [D43135611](https://our.internmc.facebook.com/intern/diff/D43135611/)

fegin committed:
[FSDP] Avoid printing incorrect warning for _get_param_to_fqns

There exist a hack for `_get_param_to_fqns` and `_apply_to_modules`. The condition for the warning of the hack is incorrect and result in overwhelming message for users. This PR fixes the issue.

The original hack is not removed. It will once the support of DMP + FSDP is deprecated.

Differential Revision: [D43135611](https://our.internmc.facebook.com/intern/diff/D43135611/)

[ghstack-poisoned]

fegin committed:
Update on "[FSDP] Avoid printing incorrect warning for _get_param_to_fqns"


There exist a hack for `_get_param_to_fqns` and `_apply_to_modules`. The condition for the warning of the hack is incorrect and result in overwhelming message for users. This PR fixes the issue.

The original hack is not removed. It will once the support of DMP + FSDP is deprecated.

Differential Revision: [D43135611](https://our.internmc.facebook.com/intern/diff/D43135611/)

[ghstack-poisoned]

rohan-varma reviewed (approved):
Thanks for the temporary fix as we work to deprecate DMP + FSDP and switch to composable APIs.

fegin committed:
Update on "[FSDP] Avoid printing incorrect warning for _get_param_to_fqns"


There exist a hack for `_get_param_to_fqns` and `_apply_to_modules`. The condition for the warning of the hack is incorrect and result in overwhelming message for users. This PR fixes the issue.

The original hack is not removed. It will once the support of DMP + FSDP is deprecated.

Differential Revision: [D43135611](https://our.internmc.facebook.com/intern/diff/D43135611/)

[ghstack-poisoned]

fegin committed:
Update on "[FSDP] Avoid printing incorrect warning for _get_param_to_fqns"


There exist a hack for `_get_param_to_fqns` and `_apply_to_modules`. The condition for the warning of the hack is incorrect and result in overwhelming message for users. This PR fixes the issue.

The original hack is not removed. It will once the support of DMP + FSDP is deprecated.

Differential Revision: [D43135611](https://our.internmc.facebook.com/intern/diff/D43135611/)

[ghstack-poisoned]

fegin commented:
@pytorchbot merge -f "The failing tests are not related."
